dependencies-path.yaml

```diff
@@ -0,0 +1,54 @@
+- dependency: "etcd"
+  refPaths:
+  - build/root/WORKSPACE
+  - cluster/gce/config-test.sh
+  - cluster/gce/manifests/etcd-empty-dir-cleanup.yaml
+  - cluster/gce/manifests/etcd.manifest
+  - cluster/gce/upgrade-aliases.sh
+  - cluster/images/etcd-empty-dir-cleanup/Makefile
+  - cluster/images/etcd/Makefile
+  - cluster/images/etcd/README.md
+  - cluster/images/etcd/migrate-if-needed.sh
+  - cmd/kubeadm/app/constants/constants.go
+  - cmd/kubeadm/app/constants/constants_test.go
+  - hack/lib/etcd.sh
+  - staging/src/k8s.io/sample-apiserver/artifacts/example/rc.yaml
+  - test/e2e/apimachinery/aggregator.go
+  - test/e2e/framework/nodes_util.go
+  - test/kubemark/start-kubemark.sh
+
+- dependency: "docker"
+  refPaths:
+  - cmd/kubeadm/app/util/system/docker_validator.go
+  - cmd/kubeadm/app/util/system/docker_validator_test.go
+
+- dependency: "golang"
+  refPaths:
+  - build/build-image/cross/Dockerfile
+  - build/build-image/cross/VERSION
+  - build/root/WORKSPACE
+  - test/images/Makefile
+
+- dependency: "cni"
+  refPaths:
+  - build/debian-hyperkube-base/Makefile
+  - build/rpms/kubeadm.spec
+  - build/rpms/kubelet.spec
+  - build/workspace.bzl
+  - cluster/gce/gci/configure.sh
+  - test/e2e_node/remote/utils.go
+
+- dependency: "coreDNS"
+  refPaths:
+  - cluster/addons/dns/coredns/coredns.yaml.base
+  - cluster/addons/dns/coredns/coredns.yaml.in
+  - cluster/addons/dns/coredns/coredns.yaml.sed
+  - cmd/kubeadm/app/constants/constants.go
+  - cmd/kubeadm/app/phases/upgrade/compute_test.go
+
+- dependency: "crictl"
```

Comment:
sadly, there is a separate here between crictl in k/k and the one in k/release WRT to debs/rpms. [1]

Reply:
under k/release is mostly how we build the artifacts, I think we should scope this file for now to dependency management across the k/k code base

Reply:
+1 to scope the file to what we actually test in the CI to be more precise.